src/lib/test-handler.js

```diff
@@ -115,11 +108,17 @@ export default class TestHandler {
           console.log(this.outputHandler.getSummaryOutput());
         }
       }
-    });
 
-    worker.on('error', (worker, error) => {
-      console.log('Error caught: ', error);
-    });
+      if (payload.exception) {
+        console.log('Error caught: ', payload.exception);
+      }
+    }.bind(this);
+
+    worker.execute(done);
+
+    worker.debugLogArray = [];
+    worker.scenario = scenario;
```

Comment:
Let's make scenario part of worker's official api rather than adding it on afterwords.